### DISCUSSION TITLE
Getting Error 400: Your request was malformed. 'bad_request' while creating customer on Paddle Billing

### DISCUSSION THREAD

**Discussion by inventionsbyhamid** [pay-rails/pay#1021] [Q&A]
Thanks for the excellent library. I am already using it with Stripe. Now adding support for Paddle Billing. I am creating a customer on Paddle to pass data-customer-id to the Paddle Checkout button using below code. I am encountering an error `Pay::PaddleBilling::Error (Error 400: Your request was malformed. 'bad_request'):`

`current_user.set_payment_processor :paddle_billing`
`current_user.payment_processor.customer
`

**Environment:**
- Pay gem version: 7.1
- Ruby version: 3.2.3
- Rails version:  7.1.3
- Operating System: MacOS Sonoma 14.5
- paddle gem: 'paddle', '~> 2.2', '>= 2.2.1'
- paddle js: https://cdn.paddle.com/paddle/v2/paddle.js

I have a pricing page where I want to show both the options, Pay with stripe or Pay with Paypal (Paddle). 

I am testing in Paddle sandbox and have the necessary env variables setup.
PADDLE_BILLING_ENVIRONMENT=sandbox
PADDLE_BILLING_API_KEY=hidden
PADDLE_BILLING_CLIENT_TOKEN=hidden
PADDLE_BILLING_SIGNING_SECRET=hidden

I might be missing something i feel, have followed whatever Paddle docs are there in pay and official Paddle dev docs.

**Comment by deanpcmad** [comment]
It's likely the customer already exists when trying to create it on Paddle. See issue #1009 as we are working to introduce nicer error messages for this.

**Reply by inventionsbyhamid** [reply]
Hi, thanks for replying. Yes it was indeed the customer already existing on Paddle. Adding a check resolved the error. For some reason the processor_id for Pay::Customer when processor is paddle_billing is not getting populated.

#<Pay::Customer id: 27, owner_type: "User", owner_id: 26, processor: "paddle_billing", processor_id: nil, default: true, data: nil, stripe_account: nil, deleted_at: nil, created_at: "2024-07-17 05:10:20.000769000 +0000", updated_at: "2024-07-17 05:12:59.597868000 +0000", payment_method_token: nil>

I think that's why pay is trying to create a new user every time when using current_user.payment_processor.customer, this works fine for stripe.

I am testing in Paddle sandbox environment. I have webhook setup via ngrok for testing with subscription, transaction and customer events enabled.

**Reply by danielschnee** [reply]
Getting the same error for completely new users.

Register a new user with devise and the run `current_user.payment_processor.customer` end with the same error:
`Error 400: Your request was malformed. 'bad_request'`

Running `Paddle::Customer.create()` directly works without any problem.
